src/components/FilterMenu/FilterMenu.js

```diff
@@ -326,127 +341,143 @@ const FilterMenu = ({
     }
   }, [autoRefresh, hidden, enableAutoRefresh, filtersStore, applyChanges])
+
+  const elementToRender = useCallback(
+    filter => {
+      switch (filter.type) {
+        case TAG_FILTER:
+          return (
+            <TagFilter
+              label={filter.label}
+              onChange={item => handleSelectOption(item, filter)}
+              page={page}
+              tagFilterOptions={tagOptions}
+              value={filtersStore[TAG_FILTER]}
+            />
+          )
+        case LABELS_FILTER:
+          return (
+            <Input
+              density='dense'
+              label={filter.label}
+              onChange={setLabels}
+              onBlur={onBlur}
+              onKeyDown={onKeyDown}
+              placeholder='key1,key2=value,...'
+              type='text'
+              value={labels}
+            />
+          )
+        case NAME_FILTER:
+          return (
+            <Input
+              density='dense'
+              label={filter.label}
+              onChange={setName}
+              onBlur={onBlur}
+              onKeyDown={onKeyDown}
+              type='text'
+              value={name}
+            />
+          )
+        case ENTITIES_FILTER:
+          return (
+            <Input
+              density='dense'
+              label={filter.label}
+              onChange={setEntities}
+              onBlur={onBlur}
+              onKeyDown={onKeyDown}
+              type='text'
+              value={entities}
+            />
+          )
+        case DATE_RANGE_TIME_FILTER:
+          return (
+            <DatePicker
+              date={filtersStore.dates.value[0]}
+              dateTo={filtersStore.dates.value[1]}
+              label={filter.label}
+              onChange={handleChangeDates}
+              type='date-range-time'
+              optionsType='past'
+            />
+          )
+        case ITERATIONS_FILTER:
+          return (
+            <CheckBox
+              item={{ label: filter.label, id: 'iter' }}
+              onChange={handleIter}
+              selectedId={filtersStore.iter}
+            />
+          )
+        case SHOW_UNTAGGED_FILTER:
+          return (
+            <CheckBox
+              className='filters-checkbox'
+              item={{ label: filter.label, id: SHOW_UNTAGGED_ITEMS }}
+              onChange={handleShowUntagged}
+              selectedId={filtersStore.showUntagged}
+            />
+          )
+        case PROJECT_FILTER:
+          return (
+            <Select
+              density='dense'
+              className=''
+              label={filter.label}
+              onClick={project => handleSelectOption(project, filter)}
+              options={filtersStore.projectOptions}
+              selectedId={filtersStore.project}
+            />
+          )
+        default:
+          return (
+            <Select
+              density='dense'
+              className={filter.type === PERIOD_FILTER ? 'period-filter' : ''}
```

Comment:
Looks like we don't use PERIOD_FILTER anymore. You can delete any logic connected to that filter

Reply:
@mavdryk Period has been removed. Added missing filter `PROJECT_FILTER` to utils since we use it here
filter.type === PROJECT_FILTER

Reply:
@ilan7empest  Don't get it. If **Period** filter was removed why we still have condition for the PERIOD_FILTER?

Reply:
@mavdryk we don't. It is outdated. Check files changes tab